### PR TITLE
File upload endpoint for BinaryFilesController

### PR DESCRIPTION
**Pull request by benwiley**
- Add upload endpoint to binary files controller.  The file type can be passed as an id or guid.  The file is uploaded and an id is returned if successful. There are various error messages for invalid types, invalid files, etc and I tried to be specific with the messages.

- Move EXIFextractor DLL from RockWeb to Rock for use in new ImageUtilities class

- Created static ImageUtilities class within Rock.Utilities.  This has the functionality previously in ImageUploader that was in RockWeb.  The move allows use of this code both in RockWeb and Rock.Rest for the upload endpoint of binary files controller.

- Reduce ImageUploader to nothing but a call to the ImageUtilities since that code was moved.

- Ignore files directory.  When debugging locally, I was setting people's profile images.  Without this ignore setting, the profile images could be committed.

Would appreciate feedback to ensure this is right before merging in!

With this implementation, steps to set a profile image:
1.) POST to api/BinaryFiles/Upload to upload file and get it's id
2.) PATCH to api/People/X to set PhotoId
3.) DELETE to api/BinaryFiles/X to delete old profile photo (if applicable)

cc @jbaxleyiii @dcs619

### CODE REVIEW

**Comment by dcs619** [comment]
Implements #1386

**Comment by jbaxleyiii** [comment]
@dcs619 @benwiley can we cherry-pick this and deploy it to alpha?

File: Rock.Rest/Controllers/BinaryFilesController.Partial.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                var binaryFileType = new BinaryFileTypeService( rockContext ).Get( binaryFileTypeId );
+
+                if ( uploadedFile == null )
+                {
+                    GenerateResponse( HttpStatusCode.BadRequest, "No file was sent" );
+                }
+
+                if ( binaryFileType == null )
+                {
+                    GenerateResponse( HttpStatusCode.InternalServerError, "There is no person image file type" );
```
Comment (azturner) [inline Rock.Rest/Controllers/BinaryFilesController.Partial.cs]: Probably should be more generic (i.e. "Invalid binary file type")

File: Rock.Rest/Controllers/BinaryFilesController.Partial.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                }
+
+                var currentUser = UserLoginService.GetCurrentUser();
+                var currentPerson = currentUser != null ? currentUser.Person : null;
+
```
Comment (azturner) [inline Rock.Rest/Controllers/BinaryFilesController.Partial.cs]: The base ApiController class has a GetPerson() method that should be used to get the person.

**Comment by azturner** [comment]
@benwiley , I Like that you moved the contents of the GetFileContentStream method in the ImageUploader to a helper class, but it looks like the ValidateFileType override got deleted too. Don't we still need that?  

Noticed too that the AppVeyor build failed for this too due to missing EXIFextractor.dll. might need to add that dll (and modify a .gitignore so it gets included).

**Comment by benwiley** [comment]
@azturner I made those changes.  Thanks for your feedback!

**Pull request merged**